activerecord/lib/active_record/relation/batches.rb

```diff
@@ -325,5 +351,12 @@ def act_on_ignored_order(error_on_ignore)
           logger.warn(ORDER_IGNORE_MESSAGE)
         end
       end
+
+      def convert_order_param_to_array(order)
+        orders = Array(order)
+        diff = Array(primary_key).size - orders.size
+        return orders if diff <= 0
+        orders + Array.new((diff), :asc)
+      end
```

Comment:
This feels overly complex, if the application didn't explicitly set an order we can assume it's `asc` without having to build an array of the defaults. Can you find a way to simplify this when we want the default case?

Reply:
It looks like there is a way for us to move the "default". handling to the `batch_order` method itself so it looks like

```ruby
Array(primary_key).zip(orders).map do |column, order| 
  o = order || DEFAULT_ORDER
  table[column].public_send(o)
end
```

We won't need much changes as we can rely on the current `zip` behavior as it fills missing values with `nil`:

```ruby
[:col1, :col2].zip([:asc]).to_h
{:col1=>:asc, :col2=>nil}
```

Though it means we'll need to find a similar approach to build the `operators` array. I'll try to leave a suggestion there as well